lib/greet.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.greet = function(name) {
+  console.log(`hello ${name}`);
+  if (!arguments.length) throw new Error('name not provided');
+  return `hello ${name}!`;
+};
+
+console.log('hello', process.argv[2]);
```

Comment:
For instance, you could have attached another method to the exports object that handles the bonus and within that method, declared a var for process.argv[2] and then pass that as the argument. Then, in index.js, you just invoke that method and keep the original greet method that accepts a name as parameter.